sycl/include/sycl/__spirv/spirv_vars.hpp

```diff
@@ -172,7 +172,7 @@ namespace __spirv {
 // Helper function templates to initialize and get vector component from SPIR-V
 // built-in variables
 #define __SPIRV_DEFINE_INIT_AND_GET_HELPERS(POSTFIX)                           \
-  template <int ID> static size_t get##POSTFIX();                              \
+  template <int ID> size_t get##POSTFIX();                                     \
```

Comment:
This is not related to the move, therefore should go in a separate PR.

Reply:
It is - if I move without this, the build fails. I'm not sure why warning isn't emitted without the move but that's not that important.

Reply:
This is a functional change.

Reply:
> I'm not sure why warning isn't emitted without the move but that's not that important.

Kind of disagree. To see different diagnostics by just renaming the file sounds important to investigate.

Reply:
I think that would be a waste of time, so if that investigation is required for making the change in this PR, I'd rather focus on other things to do.

Reply:
I checked it myself and it seems like a bug in `sycl/test/warnings/warnings.cpp` checks warnings only in `sycl` directory and ignores warnings from the headers located in other directories. Adding `--no-system-header-prefix=CL` to https://github.com/intel/llvm/blob/sycl/sycl/test/warnings/warnings.cpp#L1 reproduces the problem with existing file location.

Reply:
In that case, the warning is legit and we have to fix it here. I don't think adding the option to the test is desirable because we want to move our headers out of `CL/` instead. Also, we don't have control over OpenCL headers that reside there.

Reply:
This is a loophole in logic that defeats the purpose of the test. Headers from `sycl/` include headers from `CL/` and `std/` directories, whereas test only checks for warnings in `sycl/` directory. We can pass this test, but still have warnings in the code.